File: Hello/src/GameOfLife.scala
import scala.collection.mutable.ListBuffer
import scala.io.StdIn._
import util.control.Breaks._
class universe(aliveCellsList:List[(Int,Int)]){
  private val grid = Array.ofDim[Boolean](5,5)
  def initialize(): Unit ={
    for(i <- 0 to 4){
      for(j <- 0 to 4){
        if (aliveCellsList.contains((i,j))) grid(i)(j)=true
        else grid(i)(j)=false
      }
    }
  }
  private def countAliveNeighbours(xCoordinate:Int, yCoordinate:Int): Int ={
    var count=0
    for(i <- xCoordinate-1 to xCoordinate+1){
      for(j <- yCoordinate-1 to yCoordinate+1){
        breakable {
          if (i < 0 || i >= this.grid.length || j < 0 || j >= this.grid(0).length) break
          if ((this.grid(i)(j) != true) || (i == xCoordinate && j == yCoordinate)) break
          count+=1
        }
      }
    }
    return count
  }
  def tickTransition(): universe ={
    val temp=new universe(List())
    for(i <- 0 to this.grid.length-1){
      for(j <- 0 to this.grid(0).length-1){
        val aliveCountOfNeighbours = this.countAliveNeighbours(i, j)
        if(aliveCountOfNeighbours < 2) temp.grid(i)(j)=false
        else if (aliveCountOfNeighbours > 3) temp.grid(i)(j) = false
        else if (aliveCountOfNeighbours == 3) temp.grid(i)(j) = true
        else temp.grid(i)(j) = this.grid(i)(j)
      }
    }
    return temp
  }
  def isAlive(xCoordinate:Int, yCoordinate:Int): Boolean ={
    if(grid(xCoordinate)(yCoordinate)==true) return true
    return false
  }
}
object GameOfLife {
  def main(args: Array[String]): Unit = {
    println("Enter the number of alive cells in the seed")
    var noOfLiveCells = readInt()
    println("Enter the cordinates of the alive cells one by one")
    var aliveCellsListBuffer = ListBuffer[(Int, Int)]()
    for (i <- 0 to noOfLiveCells - 1) {
      var i = readLine().split(" ").map(_.toInt)
      var t = (i(0), i(1))
      aliveCellsListBuffer += t
    }
    val aliveCellsList = aliveCellsListBuffer.toList
    var seed = new universe(aliveCellsList)
    seed.initialize()
    for(i<- 0 to 4){
      for(j<- 0 to 4){
        if(seed.isAlive(i,j)) print("X ")
        else print("- ")
      }
      println()
    }
    println()
    var nestGenAfterTick = new universe(List())
    nestGenAfterTick.initialize()
    nestGenAfterTick = seed.tickTransition()
    for(i<- 0 to 4){
      for(j<- 0 to 4){
        if(nestGenAfterTick.isAlive(i,j)) print("X ")
        else print("- ")
      }
      println()
    }
    //commit-2 ha-it
  }
}

File: Money/GameOfLife/src/gameOfLife.java
import java.io.*;
import java.util.*;
class universe{
    public boolean[][] grid;
    public universe(int m,int n){
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                grid[i][j]=false;

            }
        }
    }
    public int count_alive(int a,int b){
        int count=0;
        for(int i=a-1;i<a+2;i++){
            for(int j=b-1;j<b+2;j++){
                if(i>=0 && i<=this.grid.length && j>=0 && j<=this.grid[0].length){
                    if(this.grid[i][j]=true && (i!=a || j!=b)){
                        count=count+1;
                    }
                    else{
                        continue;
                    }
                }
                else{
                    continue;
                }
            }
        }
        return count;
    }
    public universe tick(){
        universe temp=new universe(this.grid.length,this.grid[0].length);
        for(int i=0;i<this.grid.length;i++){
            for(int j=0;j<this.grid[0].length;j++){
                int alive_count=this.count_alive(i,j);
                if(alive_count<2){
                    temp.grid[i][j]=false;
                }
                else if(alive_count>3){
                    temp.grid[i][j]=false;
                }
                else if(alive_count==3){
                    temp.grid[i][j]=true;
                }
                else{
                    temp.grid[i][j]=this.grid[i][j];
                }
            }
        }
        return temp;
    }
}
public class gameOfLife {
    public static void main(String args[]){
        System.out.println("Enter the number of alive cells in seed");
        Scanner sc=new Scanner(System.in);
        int n;
        n=sc.nextInt();
        universe grid1=new universe(5,5);
        System.out.println("Enter the alive cells in seed one by one");
        for(int i=0;i<n;i++){
            int p=sc.nextInt();
            int q=sc.nextInt();
            grid1.grid[p][q]=true;
        }
        universe grid2=grid1.tick();
        for(int i=0;i<5;i++){
            for(int j=0;j<5;j++){
                System.out.println(grid2.grid[i][j]);
            }
        }
//hey
    }
}
